Sources/Core/Extensions/String+Extensions.swift

```diff
@@ -79,4 +79,15 @@ extension String {
 
     return MathUtility.ceil(for: size)
   }
+
+  /**
+   Returns nil is the String is empty. Returns the optional String only if it is non-empty.
+   */
+  var nonEmpty: String? {
```

Comment:
non empty kinda implies boolean result. [Nonempty](https://en.wiktionary.org/wiki/nonempty)

```suggestion
  var nonempty: String? {
```

Reply:
Yeah... I didn't love this name... do you have ideas on what would make it more clear? Really it's more like a `validatedForNonEmptiness` but that seemed too verbose... probably a dependent type would make more sense. Thoughts?